shared/devices/device-revoke/index.js

```diff
@@ -1,31 +1,49 @@
 // @flow
 import * as React from 'react'
-import {Confirm, Box, Text, Icon, type IconType} from '../../common-adapters'
-import {globalColors, globalMargins, globalStyles} from '../../styles'
+import {type DeviceType} from '../../constants/types/devices'
+import {Confirm, Box, Text, Icon, ProgressIndicator, type IconType} from '../../common-adapters'
+import {globalColors, globalMargins, globalStyles, styleSheetCreate, isMobile} from '../../styles'
 
 export type Props = {
   currentDevice: boolean,
   deviceID: string,
   endangeredTLFs: Array<string>,
-  icon: IconType,
+  type: DeviceType,
   name: string,
   onCancel: () => void,
   onSubmit: () => void,
   waiting: boolean,
 }
 
-const Header = ({name, icon}) => (
-  <Box style={{...globalStyles.flexBoxColumn, alignItems: 'center'}}>
-    <Icon type={icon} />
-    <Text type="BodySemibold" style={styleName}>
-      {name}
-    </Text>
-  </Box>
-)
+const Header = ({name, type}: {name: string, type: DeviceType}) => {
+  const icon: IconType = {
+    backup: isMobile ? 'icon-paper-key-revoke-64' : 'icon-paper-key-revoke-48',
+    desktop: isMobile ? 'icon-computer-revoke-64' : 'icon-computer-revoke-48',
+    mobile: isMobile ? 'icon-phone-revoke-64' : 'icon-phone-revoke-48',
+  }[type]
+  return (
+    <Box style={{...globalStyles.flexBoxColumn, alignItems: 'center'}}>
+      <Icon type={icon} />
+      <Text type="BodySemibold" style={styles.name}>
+        {name}
+      </Text>
+    </Box>
+  )
+}
 
-const Body = ({endangeredTLFs, name, currentDevice}) => (
+const Body = ({
```

Comment:
we're moving towards not destructuring these props as parameters, it just makes the file take a lot of vertical space for very little
instead
```
const Body  = (props: Props) => (
... this.props.waiting? ...
)
```


or if you use a prop a bunch (usually unlikely) you can just destructure that one prop
```
const Body = (props: Props) => {
  const {waiting} = props
...
}
```